src/x509/clu_config.c

```diff
@@ -74,27 +74,31 @@ static WOLFSSL_X509_EXTENSION* wolfCLU_parseBasicConstraint(char* str, int crit)
         wolfSSL_X509_EXTENSION_free(ext);
         return NULL;
     }
-
-    for (word = strtok_r(str, deli, &end); word != NULL;
-            word = strtok_r(NULL, deli, &end)) {
-        if (word != NULL && strncmp(word, "CA", strlen(word)) == 0) {
-            word = strtok_r(NULL, deli, &end);
-            if (word != NULL && strncmp(word, "TRUE", strlen(word)) == 0) {
+    
+    if (str == NULL) {
                 obj->ca = 1;
+    }
+    else {
+        for (word = strtok_r(str, deli, &end); word != NULL;
+                word = strtok_r(NULL, deli, &end)) {
+            if (word != NULL && strncmp(word, "CA", strlen(word)) == 0) {
```

Comment:
use the XSTRNCMP and X* macros wherever possible, they are abstracted for easy porting

Reply:
That code wasn't added in this PR (just moved), but I can make those changes.

Reply:
probably code I added, please do improve it with XSTRNCMP and XSTRLEN use for portability while moving it

Reply:
Done

Reply:
thanks!